Shared/Extensions/String+Extension.swift

```diff
@@ -119,6 +119,17 @@ public extension String {
         return self.trimmingCharacters(in: .whitespaces).isEmpty
     }
 
+    /// Find all matches within the given range of the string.
+    /// The resulting matches are then transformed into an array of substrings and returned
+    func match(_ regex: String) -> [String] {
```

Comment:
This seems like it would be a prime candidate for a BUNCH of tests to know it's doing exactly what we want.